types/react-calendar-timeline/index.d.ts

```diff
@@ -1,131 +1,395 @@
+        visibleTimeEnd?: Date | Moment;
+        selected?: number[];
+        sidebarWidth?: number;
+        sidebarContent?: React.ReactNode;
+        rightSidebarWidth?: number;
+        rightSidebarContent?: React.ReactNode;
+        dragSnap?: number;
+        minResizeWidth?: number;
+        lineHeight?: number;
+        itemHeightRatio?: number;
+        minZoom?: number;
+        maxZoom?: number;
+        clickTolerance?: number;
+        canMove?: boolean;
+        canChangeGroup?: boolean;
+        canResize?: false | true | 'left' | 'right' | 'both';
+        useResizeHandle?: boolean;
+        stackItems?: boolean;
+        traditionalZoom?: boolean;
+        itemTouchSendsClick?: boolean;
+        timeSteps?: TimelineTimeSteps;
+        scrollRef?: React.Ref<any>;
+        onItemDrag?(itemDragObject: OnItemDragObjectMove | OnItemDragObjectResize): void;
+        onItemMove?(itemId: number, dragTime: number, newGroupOrder: number): void;
+        onItemResize?(itemId: number, endTimeOrStartTime: number, edge: 'left' | 'right'): void;
+        onItemSelect?(itemId: number, e: any, time: number): void;
+        onItemDeselect?(e: React.SyntheticEvent): void;
+        onItemClick?(itemId: number, e: React.SyntheticEvent, time: number): void;
+        onItemDoubleClick?(itemId: number, e: React.SyntheticEvent, time: number): void;
+        onItemContextMenu?(itemId: number, e: React.SyntheticEvent, time: number): void;
+        onCanvasClick?(groupId: number, time: number, e: React.SyntheticEvent): void;
+        onCanvasDoubleClick?(group: CustomGroup, time: number, e: React.SyntheticEvent): void;
+        onCanvasContextMenu?(group: CustomGroup, time: number, e: React.SyntheticEvent): void;
+        onZoom?(timelineContext: TimelineContext): void;
+        moveResizeValidator?(
+            action: 'move' | 'resize',
+            itemId: number,
+            time: number,
+            resizeEdge: 'left' | 'right',
+        ): number;
+        onTimeChange?(
+            visibleTimeStart: number,
+            visibleTimeEnd: number,
+            updateScrollCanvas: (start: number, end: number) => void,
+        ): any;
+        onBoundsChange?(canvasTimeStart: number, canvasTimeEnd: number): any;
+        itemRenderer?: (props: ReactCalendarItemRendererProps<CustomItem>) => React.ReactNode;
+        groupRenderer?: (props: ReactCalendarGroupRendererProps<CustomGroup>) => React.ReactNode;
+        resizeDetector?: (containerResizeDetector: any) => void;
+        verticalLineClassNamesForTime?: (start: number, end: number) => string[] | undefined;
+        horizontalLineClassNamesForGroup?: (group: CustomGroup) => string[];
+
+        // Fields that are in propTypes but not documented
+        stickyHeader?: boolean;
+        headerRef?: React.Ref<any>;
+        showCursorLine?: boolean;
+        headerLabelFormats?: TimelineHeaderLabelFormat;
+        subHeaderLabelFormats?: TimelineHeaderLabelFormat;
```

Comment:
`subHeaderLabelFormats` and `headerLabelFormats` were removed from the package

Reply:
Found in master :) Kept :) But surely I'll remove :)

Reply:
I will make sure to remove them from the source code.

Reply:
@ilaiwi Take a loot at `labelFormat` at DataHeaderProps. Does it have `formatOptions` of this shape still? Or some different?

Reply:
@rip21 this is intended to be for internal use only. I had it before as a param but decided not to include it in the public API to make things more clear.

Reply:
@ilaiwi so does it mean that
```typescript
    export interface DateHeaderProps<Data> {
        style?: React.CSSProperties;
        className?: string;
        unit?: Unit | 'primaryHeader';
        labelFormat?:
            | string
            | ((
                  [startTime, endTime]: [Moment, Moment],
                  unit: Unit,
                  labelWidth: number,
                  formatOptions: TimelineHeaderLabelFormat, // <== This can be removed?
              ) => string);
        intervalRenderer?: (props?: IntervalRenderer<Data>) => React.ReactNode;
        headerData?: Data;
        children?: (props: SidebarHeaderChildrenFnProps<Data>) => React.ReactNode;
        height?: number;
    }
```

Reply:
yes, it should be removed. `short`, `medium` and `long` are very relative and might change. So removing it would be better

Reply:
@Ilaiwi  Also, this tuple `[startTime, endTime]: [Moment, Moment],` at the beginning feels a bit off. Is it tuple? Or `Interval` object instead?
{ startTime, endTime }
And whether it's Moment or it's actually `number | Date | Moment`

Reply:
Okay, I got lost a bit :) 
Can you write typing here for 
```
labelFormat?:
            | string
            | ((
                  [startTime, endTime]: [Moment, Moment],
                  unit: Unit,
                  labelWidth: number,
                  formatOptions: TimelineHeaderLabelFormat, // <== This can be removed?
              ) => string);
```
Because I'm not sure what I need to remove :)

Reply:
yes it is intended to be a tuple. is there a better way in typescript to define it as a tuple?
and yes they are moment object

Reply:
```
labelFormat?:
            | string
            | ((
                  [startTime, endTime]: [Moment, Moment],
                  unit: Unit,
                  labelWidth: number,
              ) => string);
```

Reply:
I think helper like such may help :) Have no idea why such is not included into the TS itself :)
```
type Tuple<LeftPart, RightPart> = [LeftPart, RightPart]

Tuple<Moment, Moment>
```

Thanks for the clarification :)